src/pages/index.js

```diff
@@ -1,122 +1,187 @@
-);
-const cardSection = new Section(initialCards, renderCard, cardListEl);
-cardSection.renderItems();
+const profilePopup = new PopupWithForm("#profile-avatar-modal", (url) => {
+  // handles the avatar picture submit
+  profilePopup.renderLoading(true);
+  api
+    .setUserAvatar(url)
+    .then((res) => {
+      profileAvatarFormValidator.toggleButtonState();
+      userInfo.setUserAvatar(res);
+      profilePopup.close();
+      profilePopup.renderLoading(false);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+});
+profilePopup.setEventListeners();
 
+const userInfoPopup = new PopupWithForm("#profile-edit-modal", (values) => {
+  userInfo.setUserInfo(values);
+  userInfoPopup.renderLoading(true);
+  api
+    .setUserInfo(values)
+    .then(() => {
+      editFormValidator.toggleButtonState();
+      userInfoPopup.close();
+      userInfoPopup.renderLoading(false);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+});
 userInfoPopup.setEventListeners();
-//----------------functions
 
 function renderCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
-  return card.getView();
-}
+  const card = new Card(
+    cardData,
+    constants.cardTemplate,
+
+    //handles the image click
+    function imageBtnClikc(name, link) {
+      imagePopup.open(name, link);
+    },
+    //handles the delete button click
+    function deleteBtnClick(cardInstance) {
+      cardDeleteModal.open();
+
+      cardDeleteModal.setSubmitAction(() => {
+        api.deleteCard(cardInstance.getId()).then(() => {
+          cardDeleteModal.close();
+        });
+        cardInstance.removeCard();
+
+        // cardElement = null;
+      });
+    },
+    //handles Like Button Click
+    function addLike(cardInstance) {
+      api
+        .likeCard(cardInstance.getId())
+        .then((res) => {
+          card.toggleLikeBtn();
+        })
+        .catch((err) => {
+          console.error(`Error ${err}`);
+        });
+    },
+    //removes Like Button
+    function removeLike(cardInstance) {
+      api
+        .unlikeCard(cardInstance.getId())
+        .then(() => {
+          cardInstance.toggleLikeBtn();
+        })
+        .catch((err) => {
+          console.error(`Error ${err}`);
+        });
+    }
+  );
 
-//----------------Event handlers
-function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
-  userInfoPopup.close();
+  return card.getView();
 }
 
-function handleAddCardFormSubmit(inputValues) {
-  const name = inputValues.title;
-  const link = inputValues.url;
+// EVENT LISTENERS
 
-  cardSection.addItem({ name, link });
-  addFormValidator.toggleButtonState();
-
-  cardPopup.close();
-}
-function handleImageClick(name, link) {
-  imagePopup.open(name, link);
-}
-
-// open the modal profile
-profileEditButton.addEventListener("click", () => {
+constants.addNewCardButton.addEventListener("click", () => {
+  cardPopup.open();
```

Comment:
`.toggleButtonState();` should be called after `close` because inside `close` you reset the form. Or better call it when you open modals to 100% disable the button if the form was reset after submitting.